browse/src/components/Video/ThermalVideoPlayer.vue

```diff
@@ -0,0 +1,585 @@
+              newTrack.trackId = newTrack.id;
+              this.$emit("trackSelected", newTrack);
+            }
+          }
+        });
+
+        canvas.addEventListener("mousemove", (event) => {
+          const canvasOffset = canvas.getBoundingClientRect();
+          const x = event.x - canvasOffset.x;
+          const y = event.y - canvasOffset.y;
+          const hitRect = hitTestPos(x, y);
+          // set cursor
+          canvas.style.cursor = hitRect !== null ? "pointer" : "default";
+        });
+      }
+    },
+    setTimeAndRedraw(time) {
+      this.videoJsPlayer() && this.videoJsPlayer().currentTime(time);
+      this.videoJsPlayer().play();
+    },
+    ratechange() {
+      const htmlPlayer = this.$refs.player.$refs.video;
+
+      if (htmlPlayer) {
+        localStorage.setItem("playbackrate", htmlPlayer.playbackRate);
+      }
+    },
+    seeking(event) {
+      // If the user is moving the time slider on the video then update the canvas
+      // as well so that it matches the underlying video frame.
+      if (event.type === "seeking") {
+        this.draw();
+      }
+    },
+    videoJsPlayer(): Player {
+      return this.$refs.player && this.$refs.player.player;
+    },
+    drawRectWithText(
+      context,
+      { trackIndex, trackId, rectWidth, rectHeight, x, y }
+    ) {
+      context.strokeStyle = this.colours[trackIndex % this.colours.length];
+      const selected =
+        this.currentTrack && this.currentTrack.trackId === trackId;
+
+      const lineWidth = selected ? 3 : 1;
+      const halfLineWidth = lineWidth / 2;
+      context.lineWidth = lineWidth;
+      context.strokeRect(
+        x - halfLineWidth,
+        y - halfLineWidth,
+        rectWidth + halfLineWidth,
+        rectHeight + halfLineWidth
+      );
+      if (selected) {
+        context.font = "12px Verdana";
+        context.fillStyle = "white";
+        const text = `Track ${trackIndex + 1}`;
+        const textHeight = 12;
+        const textWidth = context.measureText(text).width;
+
+        // TODO Determine if the box can be shown right at the bottom of the screen
+        // if it can then we probably need to detect this and display the text above
+        // or inside the box.
+        const textX = x + (rectWidth - textWidth);
+        const textY = y + (rectHeight + textHeight);
+        context.fillText(text, textX, textY);
+      }
+    },
+    getVideoFrameDataForAllTracksAtTime(currentTime, currentTrackOnly) {
+      // First check if the last position we got is still the current position?
+      // See if tracks are in range.
+      let tracks;
+      if (currentTrackOnly) {
+        if (this.tracks.length && this.tracks[this.currentTrack.trackIndex]) {
+          tracks = [this.tracks[this.currentTrack.trackIndex]];
+        } else {
+          tracks = [];
+        }
+      } else {
+        tracks = this.tracks;
+      }
+      const frameTime = 1 / 10;
```

Comment:
Seems like assumptions about framerate wouldn't be true for non-cptv video?

Reply:
No but didnt think we'd eve use this for cptv files, i'll add a check for the type anyway